string collsion_obj_name
---
bool success